File: tools/fix_encoding.php
<?php
// ملف لإصلاح الترميز في ملف SQL
$inputFile = 'alghazali.sql';
$outputFile = 'alghazali_fixed.sql';

// قراءة الملف الأصلي
$content = file_get_contents($inputFile);

if ($content === false) {
    die("خطأ في قراءة الملف الأصلي\n");
}

// إصلاح الترميز: تحويل من Windows-1252 إلى UTF-8
// لأن هذا هو النمط الأكثر شيوعًا لظهور هذا النوع من الأخطاء
$fixedContent = mb_convert_encoding($content, 'UTF-8', 'Windows-1252');

// إذا لم يعمل التحويل السابق، جرب تحويل من ISO-8859-1
if (strpos($fixedContent, 'Ø§') !== false) {
    $fixedContent = mb_convert_encoding($content, 'UTF-8', 'ISO-8859-1');
}

// حفظ الملف المُصَحَّح
if (file_put_contents($outputFile, $fixedContent) === false) {
    die("خطأ في حفظ الملف المُصَحَّح\n");
}

echo "تم إصلاح الترميز بنجاح!\n";
echo "الملف الأصلي: $inputFile\n";
echo "الملف المُصَحَّح: $outputFile\n";
?>

File: tools/fix_encoding.py
# -*- coding: utf-8 -*-
import os

# ملفات الإدخال والإخراج
input_file = 'alghazali.sql'
output_file = 'alghazali_fixed.sql'

# قراءة الملف باستخدام ترميز Windows-1252
with open(input_file, 'r', encoding='cp1252') as f:
    content = f.read()

# حفظ الملف باستخدام ترميز UTF-8
with open(output_file, 'w', encoding='utf-8') as f:
    f.write(content)

print(f"✅ تم إصلاح الترميز بنجاح!")
print(f"📄 الملف الأصلي: {input_file}")
print(f"📄 الملف المُصَحَّح: {output_file}")
